enos/enos-scenario-autopilot.hcl

```diff
@@ -38,13 +39,24 @@ scenario "autopilot" {
   }
 
   step "build_vault" {
-    module = matrix.builder == "crt" ? module.build_crt : module.build_local
+    module = "build_${matrix.build}"
 
     variables {
-      build_tags  = var.vault_local_build_tags != null ? var.vault_local_build_tags : local.build_tags[matrix.edition]
-      bundle_path = local.bundle_path
-      goarch      = matrix.arch
-      goos        = "linux"
+      build_tags            = var.vault_local_build_tags != null ? var.vault_local_build_tags : local.build_tags[matrix.edition]
+      bundle_path           = local.bundle_path
+      goarch                = matrix.arch
+      goos                  = "linux"
+      artifactory_host      = matrix.build == "artifactory" ? var.artifactory_host : null
+      artifactory_repo      = matrix.build == "artifactory" ? var.artifactory_repo : null
```

Comment:
These all have `null` default values. Is it necessary to check for the matrix build type for all of them if they're going to be null or set by the caller?

Reply:
💭 👍